### PR TITLE
Adding Windows Settings from PT Plugin

### PR DESCRIPTION
**Pull request by joadoumie**
This is the first pass at adding the equivalent Windows Settings plugin from PT into cmdpal.

### CODE REVIEW

File: src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Helpers/ResultHelper.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            };
+
+            // TODO GH #126 investigate tooltips
+            // AddOptionalToolTip(entry, result);
+
+            // There is a case with MMC snap-ins where we don't have .msc files fort them. Then we need to show the note for this results in subtitle too.
+            // These results have mmc.exe as command and their note property is filled.
+            if (entry.Command == "mmc.exe" && !string.IsNullOrEmpty(entry.Note))
+            {
+                result.Subtitle += $"\u0020\u0020\u002D\u0020\u0020{Resources.Note}: {entry.Note}"; // "\u0020\u0020\u002D\u0020\u0020" = "<space><space><minus><space><space>"
```
Comment (zadjii-msft) [inline src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Helpers/ResultHelper.cs:48]: 📝 these `\u0020` strings are weird to me - like, why aren't they just spaces? 

I'm sure there's a C# style reason for it I just don't know what it is.

File: src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Helpers/ResultHelper.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            }
+
+            // To not show duplicate entries we check the existing results on the list before adding the new entry. Example: Device Manager entry for Control Panel and Device Manager entry for MMC.
```
Comment (zadjii-msft) [inline src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Helpers/ResultHelper.cs:51]: frankly i'd do this _before_ you construct the `ListItem`s. May as well not bother doing you'll end up throwing out

File: src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Commands/OpenSettingsCommand.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            return false;
+        }
+#pragma warning restore CS0168, IDE0059
+    }
+
+    public override CommandResult Invoke()
+    {
+        DoOpenSettingsAction(_entry);
+
+        return CommandResult.KeepOpen();
```
Comment (zadjii-msft) [inline src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Commands/OpenSettingsCommand.cs:84]: do we want to keep open? or dismiss?

File: src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Helpers/ContextMenuHelper.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+using Microsoft.CmdPal.Extensions.Helpers;
+
+namespace Microsoft.CmdPal.Ext.WindowsSettings.Helpers;
+
+/// <summary>
+/// Helper class to easier work with context menu entries
+/// </summary>
+internal static class ContextMenuHelper
+{
+    internal static List<CommandContextItem> GetContextMenu(WindowsSetting entry)
```
Comment (zadjii-msft) [inline src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Helpers/ContextMenuHelper.cs:22]: weird to me that this isn't just a method on the class that builds the IListItems

File: src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/WindowsSettingsCommandsProvider.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    public IconDataType Icon => new(string.Empty);
+
+#pragma warning disable CA1816 // Dispose methods should call SuppressFinalize
+    public void Dispose() => throw new NotImplementedException();
+#pragma warning restore CA1816 // Dispose methods should call SuppressFinalize
+
+    public IListItem[] TopLevelCommands()
+    {
+        return [
+            new ListItem(new WindowsSettingsListPage(_windowsSettings))
```
[the remark this answers is not in the document]
Comment (joadoumie) [reply src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/WindowsSettingsCommandsProvider.cs:40]: Well that's incredibly fair -- I didn't think of that... at all. I think somewhere in the back of my mind I just was thinking it was still going to be the same allocations every time. 

How often is ```TopLevelCommands``` called? -- interesting to think about the implications. Fixing in my next commit.

[the remark this answers is not in the document]
Comment (joadoumie) [reply src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/WindowsSettingsCommandsProvider.cs:40]: What does the extension lifecycle look like? Are we going to load extensions when a command is triggered? I don't know if we've finished outlining the extension lifecycle.

In the case that we are loading extensions on the invocation of a command, does it make a difference if we create new objects in ```TopLevelCommands()```? Just a thought (and my thought process may make no sense).

File: src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Pages/WindowsSettingsListPage.cs [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+namespace Microsoft.CmdPal.Ext.WindowsSettings;
+
+internal sealed partial class WindowsSettingsListPage : DynamicListPage
+{
+    private readonly string _defaultIconPath;
+    private readonly Classes.WindowsSettings _windowsSettings;
+
+    public WindowsSettingsListPage(Classes.WindowsSettings windowsSettings)
+    {
+        Icon = new(string.Empty);
```
[the remark this answers is not in the document]
Comment (joadoumie) [reply src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Pages/WindowsSettingsListPage.cs:31]: I actually need to fix the icon story for ALL plugin ports. I am waiting for #78 to be finalized (specifically the light/dark theme stuff).

File: src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Helpers/ResultHelper.cs (hunk 1 above)
Comment (joadoumie) [reply src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Helpers/ResultHelper.cs:48]: I fully just copied the original PT plugin logic for this part and didn't want to modify things unless necessary.

File: src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Commands/OpenSettingsCommand.cs (hunk 3 above)
Comment (joadoumie) [reply src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Commands/OpenSettingsCommand.cs:84]: dismiss makes more sense. good catch.

File: src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Helpers/ContextMenuHelper.cs (hunk 4 above)
Comment (joadoumie) [reply src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Helpers/ContextMenuHelper.cs:22]: I took the same approach the OG plugin had -- I can move things around.

File: src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/WindowsSettingsCommandsProvider.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (zadjii-msft) [reply src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/WindowsSettingsCommandsProvider.cs]: Good questions - ones I'l try and document better in the next couple weeks here. 

Right now I'm pretty sure `TopLevelCommands` is called every time we come back to the Home page - so on cold launch and every time we're dismissed. It'll be more efficient in the future (once I document how)

**Review by zadjii-msft — APPROVED** [review]

**Comment by zadjii-msft** [comment]
@joadoumie I'm merging this, so I can merge the `ISection` removal in #129

**Pull request merged**